### PR TITLE
Copied projects into Play 2.5.10 compatible versions.

### PR DESCRIPTION
**Pull request by jeffmay**
@htmldoug @vodopis @patrick-bohan

- Copied project code to create Play 2.5.10 version
- Upgraded sbt
- Added coursier

I decided to copy everything over. I figure it will be easier to maintain by copying and pasting features that are relevant to both until everyone makes the switch to Play 2.5 than trying to employ the right combination of sbt magic and source code compatibility hackery.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -54,19 +57,39 @@ lazy val common = commonRootSettings ++ Seq(
...
-    "com.typesafe.play" %% "play-json" % playJsonVersion
+    "com.typesafe.play" %% "play-json" % playVersion
   )
 )
 
-lazy val playJsonTests = project in file("playJsonTests") settings(common: _*) settings (
-  name := "play-json-tests",
+lazy val `play-json-ops-23` = project in file("play-json-ops-23") settings (
+  playJsonOpsCommon(PlayJsonVersion._23),
+  name := "play-json-ops",
```
Comment (jeffmay) [inline build.sbt:68]: I'm keeping the old name for Play 2.3.10 for backwards compatibility.

File: playJsonOps/src/main/scala/play/api/libs/json/ops/JsonTransform.scala [hunk 2]
```diff
@@ -1,6 +1,5 @@
 package play.api.libs.json.ops
 
-import play.api.libs.json.ops.JsValueImplicits._
```
Comment (jeffmay) [inline playJsonOps/src/main/scala/play/api/libs/json/ops/JsonTransform.scala:3]: Unused import.

**Comment by jeffmay** [comment]
I decided to copy everything over, since there are some source code incompatibilities (only a few). I figure that it would be just as easy to copy and paste code changes than to try and figure out the right combination of sbt magic and source code compatibility management to share the same code.

File: build.sbt [hunk 3]
```diff
@@ -1,20 +1,23 @@
 
 lazy val commonRootSettings = Seq(
-  version := "1.4.0",
+  version := "1.5.0",
   scalaVersion := "2.11.8",
-  crossScalaVersions := Seq("2.11.8", "2.10.6"),
```
Comment (jeffmay) [inline build.sbt:5]: Only Play 2.3 is cross compiled, since Play 2.5 does not have a Scala 2.10 or Scala 2.12 version.

File: project/plugins.sbt [hunk 4]
```diff
@@ -5,9 +5,7 @@ resolvers += Resolver.url(
   url("http://dl.bintray.com/content/sbt/sbt-plugin-releases"))(
     Resolver.ivyStylePatterns)
 
+addSbtPlugin("io.get-coursier" % "sbt-coursier" % "1.0.0-M15")
```
Comment (jeffmay) [inline project/plugins.sbt:8]: Coursier is cool. I'm using it for personal projects.

**Comment by jeffmay** [comment]
I'm ignoring the coverage decrease for this PR. I don't want to introduce too many changes.

**Review by htmldoug — APPROVED** [review]

**Pull request merged**